BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -249,18 +255,31 @@ public void SavePost(Post post)
         public Post CollectPostInfo(string expectedFilePath)
         {
             XDocument doc = LoadPostXml(expectedFilePath);
-            Post post = new Post
+            Post post = new Post();
+            if (doc.Root.Element("Id") != null && !doc.Root.Element("Id").IsEmpty)
             {
-                Id = Convert.ToInt32(doc.Root.Element("Id").Value),
-                Slug = doc.Root.Element("Slug").Value,
-                Title = doc.Root.Element("Title").Value,
-                Body = doc.Root.Element("Body").Value,
-                PubDate = DateTimeOffset.Parse(doc.Root.Element("PubDate").Value),
-                LastModified = DateTimeOffset.Parse(doc.Root.Element("LastModified").Value),
-                IsPublic = Convert.ToBoolean(doc.Root.Element("IsPublic").Value),
-                IsDeleted = Convert.ToBoolean(doc.Root.Element("IsDeleted").Value),
-                Excerpt = doc.Root.Element("Excerpt").Value,
-            };
+                post.Id = Convert.ToInt32(doc.Root.Element("Id").Value);
+            }
+            else
+            {
+                SetId(post);
+            }
+            post.Slug = (doc.Root.Element("Slug") != null && !doc.Root.Element("Slug").IsEmpty) ?

```

Comment:
These lines are really verbose, maybe we can make them more concise?  This will also save you from searching the document 3 times for each element (`doc.Root.Element()` is repeated 3 times on each line).
For example, maybe a helper method:

    private static string GetValue(XElement e, string defaultValue)
    {
        if (e != null && !e.IsEmpty)
        { return e.Value; }
        else
        { return defaultValue; }
    }

    GetValue(doc.Root.Element("Slug"), "")
    GetValue(doc.Root.Element("IsPublic"), true.ToString())
    // etc